src/EnergyPlus/Material.cc

```diff
@@ -2668,45 +2668,12 @@ void GetMaterialData(EnergyPlusData &state, bool &ErrorsFound) // set to true if
         }
 
         if ((mat->LayerType == TARCOGParams::TARCOGLayerType::VENETBLIND_HORIZ) ||
-            (mat->LayerType == TARCOGParams::TARCOGLayerType::VENETBLIND_HORIZ)) {
-            if (s_ipsc->rNumericArgs(11) <= 0.0) {
-                ErrorsFound = true;
-                ShowSevereCustomMessage(
-                    state, eoh, format("{} must be >0, entered value = {:.2R}", s_ipsc->cNumericFieldNames(11), s_ipsc->rNumericArgs(11)));
-            }
-
-            if (s_ipsc->rNumericArgs(12) <= 0.0) {
-                ErrorsFound = true;
-                ShowSevereCustomMessage(
-                    state, eoh, format("{} must be >0, entered value = {:.2R}", s_ipsc->cNumericFieldNames(12), s_ipsc->rNumericArgs(12)));
-            }
-
-            if (s_ipsc->rNumericArgs(13) <= 0.0) {
-                ErrorsFound = true;
-                ShowSevereCustomMessage(
-                    state, eoh, format("{} must be >0, entered value = {:.2R}", s_ipsc->cNumericFieldNames(13), s_ipsc->rNumericArgs(13)));
-            }
-
-            if ((s_ipsc->rNumericArgs(14) < -90.0) || (s_ipsc->rNumericArgs(14) > 90.0)) {
-                ErrorsFound = true;
-                ShowSevereCustomMessage(
-                    state,
-                    eoh,
-                    format("{} must be >=-90 and <=90, entered value = {:.2R}", s_ipsc->cNumericFieldNames(14), s_ipsc->rNumericArgs(14)));
-            }
-
-            if (s_ipsc->rNumericArgs(15) <= 0.0) {
-                ErrorsFound = true;
-                ShowSevereCustomMessage(
-                    state, eoh, format("{} must be >0, entered value = {:.2R}", s_ipsc->cNumericFieldNames(15), s_ipsc->rNumericArgs(15)));
-            }
-
-            if ((s_ipsc->rNumericArgs(16) < 0.0) ||
-                ((s_ipsc->rNumericArgs(16) > 0.0) && (s_ipsc->rNumericArgs(16) < (s_ipsc->rNumericArgs(11) / 2)))) {
+            (mat->LayerType == TARCOGParams::TARCOGLayerType::VENETBLIND_VERT)) {
+            if ((s_ipsc->rNumericArgs(16) > 0.0) && (s_ipsc->rNumericArgs(16) < (s_ipsc->rNumericArgs(11) / 2))) {
```

Comment:
[src/EnergyPlus/Material.cc:2670]:(style),[duplicateExpression],Same expression on both sides of '||'.

These other checks are caught by InputProcessor and are never executed. For example if I make slat thickness < 0 this input processor warning is reported.

    ** Severe  ** <root>[WindowMaterial:ComplexShade][Shade_14_Layer][slat_thickness] - "-0.001000" - Expected number greater than 0.000000
    **  Fatal  ** Errors occurred on processing input file. Preceding condition(s) cause termination.

Reply:
Yes!  There is so much duplicated between the IDD/input processor range checks and then inaccessible checks in the get input routines.  I love it.